src/javascript/crypto/e2e/openpgp/pgpmime/utils.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ * Copyright 2015 Google Inc. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Helper utils for building/parsing PGP/MIME emails.
+ * @author yzhu@yahoo-inc.com (Yan Zhu)
+ */
+
+goog.provide('e2e.openpgp.pgpmime.Utils');
+
+goog.require('e2e.openpgp.pgpmime.Constants');
+goog.require('e2e.openpgp.pgpmime.Text');
+
+goog.require('goog.array');
+goog.require('goog.object');
+goog.require('goog.string');
+
+goog.scope(function() {
+var pgpmime = e2e.openpgp.pgpmime;
+var constants = pgpmime.Constants;
+var utils = pgpmime.Utils;
+
+
+/**
+ * Parses a header value string. Ex: 'multipart/mixed; boundary="foo"'
+ * @param {string} text The string to parse
+ * @return {e2e.openpgp.pgpmime.types.HeaderValue}
+ */
+e2e.openpgp.pgpmime.Utils.parseHeaderValue = function(text) {
+  var parts = text.split('; ');
+  var firstPart = parts.shift();
+
+  // Normalize value to lowercase since it's case insensitive
+  var value = goog.string.stripQuotes(firstPart.toLowerCase().trim(), '"');
+
+  var params = {};
+  goog.array.forEach(parts, function(part) {
+    // Ex: 'protocol=application/pgp-encrypted'
+    var paramParts = part.split('=');
```

Comment:
`paramParts = goog.string.splitLimit(part, '=', 2)`. 

Otherwise you wouldn't accept valid `param="a=value=with=equals"`

Later on you need to check if the parameter name and value are correct (i.e. contain only allowed characters). Parameter value allows for more characters (incl. `=`), but only if it's quoted, so you need to have a code branch for this too. See RFC 2045 5.1.